autonomy/cli/fetch.py

```diff
@@ -87,6 +88,12 @@ def fetch(
             do_fetch(ctx, public_id, alias)
         else:
             fetch_service(ctx, public_id)
+    except NotAnAgentPackage as e:
+        raise click.ClickException(
+            "Downloaded packages is not an agent package, "
+            "if you intend to download a service please use `--service` flag "
+            "or check the hash"
+        ) from e
```

Comment:
When we get to this exception, should we revert the registry status to the original (i.e., delete the downloaded package)? Also, there some coverage gap there.

Reply:
We already delete the downloaded package on failure https://github.com/valory-xyz/open-aea/blob/beadc9f172e4c331e4e95e73d17bfb4ddbd60d38/aea/cli/fetch.py#L127

Reply:
The tests for this failure are on open-aea, not here. Here we only deal with `service` related logic